src/main/java/org/isf/exa/gui/ExamEdit.java

```diff
@@ -249,49 +241,47 @@ private JButton getOkButton() {
 				okButton = new JButton();
 				okButton.setText(MessageBundle.getMessage("angal.common.ok"));  // Generated
 	            okButton.setMnemonic(KeyEvent.VK_O);
-				okButton.addActionListener(new java.awt.event.ActionListener() {
-					public void actionPerformed(java.awt.event.ActionEvent e) {
-						if((codeTextField.getText().trim().equals(""))||(descriptionTextField.getText().trim().equals(""))){
-							JOptionPane.showMessageDialog(null, MessageBundle.getMessage("angal.exa.pleaseinsertcodeoranddescription"));
-						}
-						else{
-							int procedure = Integer.parseInt(procComboBox.getSelectedItem().toString());
-							
-							exam.setExamtype((ExamType)typeComboBox.getSelectedItem());
-							exam.setDescription(descriptionTextField.getText());
-							
-							exam.setCode(codeTextField.getText().toUpperCase());
-							exam.setDefaultResult(defTextField.getText().toUpperCase());
-							exam.setProcedure(procedure);
-							
-							boolean result = false;
-							if (insert) {
-								try {
-									if (manager.isKeyPresent(exam)) {
-										JOptionPane.showMessageDialog(ExamEdit.this, MessageBundle.getMessage("angal.exa.changethecodebecauseisalreadyinuse"));
-										return;
-									}
-								} catch (OHServiceException e1) {
-									OHServiceExceptionUtil.showMessages(e1);
-								}
-								try {
-									result = manager.newExam(exam);
-									if (result) fireExamInserted();
-								} catch (OHServiceException e1) {
-									OHServiceExceptionUtil.showMessages(e1);
-								}
-							} else {
-								try {
-									result = manager.updateExam(exam);
-									if (result) fireExamUpdated();
-								} catch (OHServiceException e1) {
-									OHServiceExceptionUtil.showMessages(e1);
+				okButton.addActionListener(e -> {
+					if((codeTextField.getText().trim().equals(""))||(descriptionTextField.getText().trim().equals(""))){
```

Comment:
The `.equals("")` can be replaced by `isEmpty()` (twice); the line could use some additional spacing too.

Reply:
again part of the old code moved to anonymous function (look at line 254 from the red part of diff)